nerdlets/tag-improver-nerdlet/index.js

```diff
@@ -307,13 +343,32 @@ export default class TagVisualizer extends React.Component {
       () => {
         const { tagHierarchy, mandatoryTagCount } = this.state;
         const { updateEntityTagCompliance } = this;
-        Object.entries(tagHierarchy).forEach(tagKey => {
-          Object.values(tagKey[1]).forEach(tagValue => {
-            tagValue.forEach(entity => {
+
+        // check if policy was changed
+        if (JSON.stringify(policy) !== JSON.stringify(prevPolicy)) {
```

Comment:
This is subject to false positives since property order isn't guaranteed consistent. We could use [_.isEqual](https://www.npmjs.com/package/lodash.isequal) but I hate pulling in new dependencies if we don't have to. There's potential for UI stutter while you recompute the compliance data on large entity sets. Any other impact from false positives on this codepath?

Reply:
I modified the if statement to sort the policy and prevPolicy before comparing
```
if (
          JSON.stringify(sortedPolicy(policy)) !==
          JSON.stringify(sortedPolicy(prevPolicy))
        ) {
          ...
```